### PR TITLE
Backport 

### PR DESCRIPTION
**Pull request by uwej711**
This makes the commands sf 2.3 compatible (not tested with any newer sf version)

### CODE REVIEW

**Comment by dantleech** [comment]
Sorry, looks like this got lost in my inbox. This would be fine, the tests failed for some unknown reason, I just restarted.

**Comment by dantleech** [comment]
Hmm:

````
PHP Warning:  preg_match() expects parameter 2 to be string, object given in /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php on line 622
PHP Stack trace:
PHP   1. {main}() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony-cmf/testing/bin/console:0
PHP   2. Symfony\Component\Console\Application->run() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony-cmf/testing/bin/console:17
PHP   3. Symfony\Bundle\FrameworkBundle\Console\Application->doRun() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:121
PHP   4. Symfony\Bundle\FrameworkBundle\Console\Application->registerCommands() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:73
PHP   5. Symfony\Component\DependencyInjection\Container->get() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:105
PHP   6. appPhpcrDebugProjectContainer->getPhpcrMigrations_Command_MigrateService() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:326
PHP   7. Symfony\Component\Console\Command\Command->__construct() /home/travis/build/dantleech/phpcr-migrations-bundle/Tests/Resources/app/cache/appPhpcrDebugProjectContainer.php:1576
PHP   8. Symfony\Component\Console\Command\Command->setName() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:61
PHP   9. Symfony\Component\Console\Command\Command->validateName() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:407
PHP  10. preg_match() /home/travis/build/dantleech/phpcr-migrations-bundle/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php:622
````

**Comment by uwej711** [comment]
Lets see now, there was still a command as service definition left.

**Comment by dantleech** [comment]
Looks like the tests are still retrieving the commands from the container

**Comment by uwej711** [comment]
I will try to look at the changes needed for the tests tonight.

**Comment by uwej711** [comment]
the tests seem to require an apps/migrations directory, running them locally with the directory present works without failure. not sure whether this is related to my changes.

File: Command/StatusCommand.php [hunk 1]
```diff
@@ -70,7 +63,7 @@ public function execute(InputInterface $input, OutputInterface $output)
             ));
         }
 
-        $table->render();
+        $table->render($output);
```
Comment (dantleech) [inline Command/StatusCommand.php]: `TableHelper` is deprecated and will be removed in 3.0 - we should instantiate it only if `Table` does not exist.

**Comment by dantleech** [comment]
Not sure the `migrations` directory is required *now*. But we can simply remove the path configuration from the [config](https://github.com/dantleech/phpcr-migrations-bundle/blob/master/Tests/Resources/app/config/config.yml#L3)

File: Command/StatusCommand.php (hunk 1 above)
Comment (dantleech) [inline Command/StatusCommand.php]: Something like

````php
if (class_exists('Symfony\Component\Console\Table`) {
  // ...
} else {
  // ...
}
````

or we can keep the deprecation and just make a new release for 3.0 which drops 2.3 support

**Comment by dantleech** [comment]
As noted above the `TableHelper` will be dropped in `3.0`, we could fork the logic depending on if it exists or not, or make a new release for Symfony 3.0 (which is planned for release shortly after the EOL of Symfony 2.3) - at which point we may as well revert this PR.

So this would become version 1.0 with Sf2.3 support, and we would release 2.0 without Sf2.3 support.

**Comment by uwej711** [comment]
Tests should be ok now. For the TableHelper I would opt for a 2.0 version and require Symfony 2.4 in composer.json then.

**Comment by dantleech** [comment]
Merged, thanks uwe

**Comment by dantleech** [comment]
@uwej711 are you still using SF 2.3 ? Would like to bump the min requirements.

**Comment by uwej711** [comment]
Yes we are, but no problem, we need to update to 2.7/2.8 anyway soon.

Cheers
Uwe

**Comment by dantleech** [comment]
ok, I will tag a new release tonight probably, but you can
anchor to 0.1

**Pull request merged**